src/lib.rs

```diff
@@ -99,9 +97,276 @@ use std::fmt;
 use std::iter::repeat;
 use std::result;
 
+/// A description of the options that a program can handle
+pub struct Options {
+    grps: Vec<OptGroup>
+}
+
+impl Options {
+    /// Create a blank set of options
+    pub fn new() -> Options {
+        Options {
+            grps: Vec::new()
+        }
+    }
+
+    /// Create a generic option group, stating all parameters explicitly
+    pub fn add_opt<'a>(&'a mut self, short_name: &str, long_name: &str, desc: &str,
```

Comment:
Yeah for example all of the [Command methods](http://doc.rust-lang.org/std/io/process/struct.Command.html) are just "foo" instead of "add_foo" (even for adding arguments).